win_etw_macros/src/lib.rs

```diff
@@ -668,32 +667,35 @@ fn create_register_provider_traits(
     provider_name: &str,
     provider_group_guid: Option<&Uuid>,
 ) -> TokenStream {
-    fn align2(v: &mut Vec<u8>) {
-        if v.len() % 2 != 0 {
-            v.push(0);
-        }
-    }
-
     let mut traits_bytes: Vec<u8> = Vec::new();
     traits_bytes.push(0); // reserve space for TraitsSize (u16)
     traits_bytes.push(0);
 
     traits_bytes.extend_from_slice(provider_name.as_bytes());
     traits_bytes.push(0);
-    align2(&mut traits_bytes);
 
     if let Some(provider_group_guid) = provider_group_guid {
         // Add trait for provider guid
         let provider_guid_trait_offset = traits_bytes.len();
         traits_bytes.push(0); // reserve space for TraitSize (u16)
         traits_bytes.push(0);
         traits_bytes.push(ETW_PROVIDER_TRAIT_TYPE_GROUP);
-        traits_bytes.extend_from_slice(provider_group_guid.as_bytes());
+
+        let (data1, data2, data3, data4) = provider_group_guid.to_fields_le();
```

Comment:
Ahh, so I had the repr of the UUID wrong.

Can this just be `traits_bytes.extend_from_slice(&provider_group_guid.to_bytes_le());`?

If not, can we move the guts of this to a separate method, like:

```
fn uuid_to_bytes_le(g: &Guid) -> [u8; 16] {
    let (d1, d2, d3, d4) = g.to_fields_le();
    let d1 = d1.to_be_bytes();
    let d2 = d2.to_be_bytes();
    let d3 = d3.to_be_bytes();
    // d4 is already bytes
    [
        d1[0], d1[1], d1[2], d1[3],
        d2[0], d2[1],
        d3[0], d3[1],
        d4[0], d4[1], d4[2], d4[3], d4[4], d4[5], d4[6], d4[7],
    ]
  }
}

...
   traits_bytes.push(uuid_to_le_bytes(&provider_group_guid));
```